alica_engine/include/engine/IAlicaTrace.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <amr_interfaces/TraceContext.h>
+
+namespace alica
+{
+
+class IAlicaTrace
+{
+public:
+    virtual ~IAlicaTrace() = default;
+    IAlicaTrace(const std::string& opName, std::optional<const amr_interfaces::TraceContext> parent = std::nullopt);
+    IAlicaTrace(IAlicaTrace&& other) = default;
+    IAlicaTrace& operator=(IAlicaTrace&& other) = default;
+
+//    void setTag(const std::string& key, const RawTraceValue& value);
```

Comment:
the interface can have 4 pure virtual functions:
```
virtual void setTag(const std::string& key, const std::string& value) = 0;
virtual void setLog(const std::string& key, const std::string& value) = 0;
virtual void markError(const std::string& description) = 0;
std::string context() const = 0;
```
Ideally setTag & setLog should be able to take ints, doubles etc. as values, but can figure that out later. string should be fine for now